impl/container/src/main/java/io/serverlessworkflow/impl/container/executors/ContainerRunner.java

```diff
@@ -21,115 +21,122 @@
 import com.github.dockerjava.api.command.CreateContainerCmd;
 import com.github.dockerjava.api.command.CreateContainerResponse;
 import com.github.dockerjava.api.command.WaitContainerResultCallback;
-import com.github.dockerjava.api.exception.DockerClientException;
 import com.github.dockerjava.api.exception.NotFoundException;
 import com.github.dockerjava.core.DefaultDockerClientConfig;
 import com.github.dockerjava.core.DockerClientImpl;
 import com.github.dockerjava.core.NameParser;
 import com.github.dockerjava.httpclient5.ApacheDockerHttpClient;
 import io.serverlessworkflow.api.types.Container;
+import io.serverlessworkflow.api.types.ContainerLifetime;
+import io.serverlessworkflow.api.types.TimeoutAfter;
 import io.serverlessworkflow.impl.TaskContext;
 import io.serverlessworkflow.impl.WorkflowContext;
 import io.serverlessworkflow.impl.WorkflowDefinition;
 import io.serverlessworkflow.impl.WorkflowModel;
 import io.serverlessworkflow.impl.WorkflowUtils;
 import io.serverlessworkflow.impl.WorkflowValueResolver;
+import java.io.IOException;
 import java.time.Duration;
 import java.util.ArrayList;
-import java.util.List;
+import java.util.Collection;
+import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.TimeUnit;
-import java.util.function.Function;
 
 class ContainerRunner {
 
   private static final DefaultDockerClientConfig DEFAULT_CONFIG =
       DefaultDockerClientConfig.createDefaultConfigBuilder().build();
 
-  private static final DockerClient dockerClient =
-      DockerClientImpl.getInstance(
-          DEFAULT_CONFIG,
-          new ApacheDockerHttpClient.Builder().dockerHost(DEFAULT_CONFIG.getDockerHost()).build());
+  private static class DockerClientHolder {
+    private static final DockerClient dockerClient =
+        DockerClientImpl.getInstance(
+            DEFAULT_CONFIG,
+            new ApacheDockerHttpClient.Builder()
+                .dockerHost(DEFAULT_CONFIG.getDockerHost())
+                .build());
+  }
 
-  private final CreateContainerCmd createContainerCmd;
-  private final Container container;
-  private final List<ContainerPropertySetter> propertySetters;
-  private final WorkflowDefinition definition;
+  private final Collection<ContainerPropertySetter> propertySetters;
+  private final Optional<WorkflowValueResolver<Duration>> timeout;
+  private final ContainerCleanupPolicy policy;
```

Comment:
The only information from the Container task that is needed at runtime (once the ContainerPropertySetter implementations are changed) are the policy and container image name